aws/resource_aws_emr_instance_fleet.go

```diff
@@ -0,0 +1,193 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsEMRInstanceFleet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEMRInstanceFleetCreate,
+		Read:   resourceAwsEMRInstanceFleetRead,
+		Update: resourceAwsEMRInstanceFleetUpdate,
+		Delete: resourceAwsEMRInstanceFleetDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_fleet_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(emr.InstanceFleetType_Values(), false),
+			},
+			"instance_fleet": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem:     InstanceFleetConfigSchema(),
```

Comment:
When flattening this schema, this resource should not share the exact same schema used in `aws_emr_cluster` -- this resource supports updates so `ForceNew` must not be specified on attributes that support in-place updates.

Reply:
can we share the same schema?
the only thing can update in the resource is  target capacity.
I can update the aws_emr_cluster later to add the update function later.

Reply:
We typically prefer to _not_ share schema between resources since updates may occur in one resource, but not another (either intentionally or accidentally), which has been a source of many bugs in the past. For now though, re-using the nested schema is likely okay, since we would be sharing the handling functions anyways.

Reply:
yes, I just don't want to copy/past those functions.

Reply:
Hi @c4po 👋  To clarify here, we are expecting the children attributes of the instance fleet configuration, e.g.

* `instance_type_configs`
* `launch_specifications`
* `name`
* `target_on_demand_capacity`
* `target_spot_capacity`

To exist at the root level of this new resource. This will match the setup of the `aws_emr_instance_group` resource. We do not need the extra `task_instance_fleet` configuration block to contain all those attributes. 👍